### PR TITLE
docs - init deployment rewrite

### LINKED ISSUE #3953: docs - update the deployment section

### PR DESCRIPTION
**Pull request by thisisshi**
closes #3953 

deployment guide beginnings-- currently somewhat abstract but it might be worth doing a comprehensive step by step tutorial?

### CODE REVIEW

**Comment by kapilt** [comment]
might also be worth referencing some blog posts that have alternative detailed deployments, like this one jantman did using jenkins, template preprocessing, dead letter queue setups, monitoring, etc.
https://blog.jasonantman.com/2017/10/cloud-custodian-architecture-deployment-and-policy-preprocessing/

File: docs/source/overview/deployment.rst [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    trigger:
+    - master
+    
+    jobs: 
+      - job: 'Validate'
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        strategy:
+          matrix:
```
Comment (kapilt) [inline docs/source/overview/deployment.rst]: don't really need a matrix strategy here, we're just using a single python version

File: docs/source/overview/deployment.rst [hunk 2]
```diff
@@ -1,7 +1,91 @@
...
+tool like Github. This allows us to version and enable collaboration through
+git pull requests and issues.
+
+.. code-block:: bash
+ 
+    $ git init
+
+Next, enable a CI webhook back to your CI system of choice when pull requests
+targeting your master branch are opened or updated. This allows us to constantly
+test and validate the policies that are being modified. You can use the tool
```
[the remark this answers is not in the document]
Comment (kapilt) [reply docs/source/overview/deployment.rst]: while there's now more policystream content in advanced, the reference here should still get removed imo, so it doesn't detract from the getting started flow as its advanced.

File: docs/source/overview/deployment.rst [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+Setting up the automated dryrun of policies is left as an exercise to the user-- this
+requires hosting your cloud authentication tokens inside of a CI system or hosting your
+own CI system and using Managed Service Identities (Azure) or Instance Profiles (AWS).
+
+It's important to verify that the results of the dryrun match your expectations. Custodian
+is a very powerful tool that will do exactly what you tell it to do! In this case, you should
+always "measure twice, cut once".
+
+
+Advanced Continuous Integration Tips
```
Comment (kapilt) [inline docs/source/overview/deployment.rst]: lets move this to the bottom, I'd like the user to get through a complete working setup before they look at optimizing

File: docs/source/overview/deployment.rst [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+Once you have Cloud Custodian installed, download your policies that you created
+in the :ref:`compliance_as_code` section. If using git, just simply do a ``git clone``::
+
+    $ git clone <repository-url>
+
+You now have your policies and custodian available on the instance. Typically, policies
+that query the extant resources in the account/project/subscription should be run
+on a regular basis to ensure that resources are constantly compliant. To do this you
+can simply set up a cron job to run custodian on a set cadence.
+
```
Comment (kapilt) [inline docs/source/overview/deployment.rst]: we should add a monitoring section here

**Comment by kapilt** [comment]
This is looking really good overall, thanks

File: docs/source/overview/deployment.rst [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+To run Cloud Custodian against your account, you will need an IAM role with appropriate
+permissions. Depending on the scope of the policy, these permissions may differ from policy
+to policy. For a baseline, the managed read only policies in each of the respective cloud
+providers will be enough to dryrun your policies. Actions will require additional IAM
+permissions which should be added at your discretion.
+
+For serverless policies, Custodian will need the corresponding permissions to provision
+serverless functions.
+
+In AWS, you will need ReadOnly access as well as the following permissions:
```
Comment (thisisshi) [inline docs/source/overview/deployment.rst]: I'd like to see some more examples for other clouds here, i'm just not familiar with them

Comment (thisisshi) [reply docs/source/overview/deployment.rst]: @aluong can you follow up on this in another pr?

**Pull request merged**